main/main.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/stackrox/k8s-istio-cve-pusher/utils"
+)
+
+const (
+	logDataFeedName      = "data feed name"
+	kubernetesBucketDir  = "k8s"
+	istioBucketDir       = "istio"
+	cveFileExt           = ".json"
+	k8sCveFileName       = "NVDk8sCVEs.json"
+	k8sCveHashFileName   = "k8sCVEsHash"
+	istioCveFileName     = "NVDistioCVEs.json"
+	istioCveHashFileName = "istioCVEsHash"
+)
+
+type component int
+
+const (
+	k8s = iota
+	istio
+)
+
+type fileAndBucketPaths struct {
+	cveFileNameWithPath     string
+	cveHashFileNameWithPath string
+	cveBucketSubPath        string
+	cveHashBucketSubPath    string
+}
+
+func main() {
+	if err := runCmd(); err != nil {
+		log.Fatal(err)
+		os.Exit(1)
+	}
+}
+
+func runCmd() error {
+	path, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+
+	feedReader, err := getDataFeeds(path)
```

Comment:
This is true, but given he's done it this way already, I would say it's not worth the time to refactor at this point. (It's not without complexity to do the refactor since you need to be careful not to close the response body too early.)

Reply:
Agreed. Just an observation, and shouldn't block this PR.